server/service/service_osquery.go

```diff
@@ -419,11 +420,59 @@ const hostDistributedQueryPrefix = "fleet_distributed_query_"
 
 // detailQueriesForHost returns the map of detail+additional queries that should be executed by
 // osqueryd to fill in the host details.
+
+var (
+	lastCheck time.Time
+	rate      int = 10
+)
+
 func (svc *Service) detailQueriesForHost(ctx context.Context, host fleet.Host) (map[string]string, error) {
```

Comment:
This is "rate limiting distributed query writes" *indirectly*, right? (just clarifying what the title says).

Reply:
Yes, it's really more like "rate limiting sending queries".

Reply:
(the way it works is by delaying _sending_ the queries that when _received_ take a lot of processing for MySQL)